File: 프로그래머스/lv3/12907. 거스름돈/거스름돈.java
class Solution {
    public int solution(int n, int[] money) {
        int[] dp = new int[n + 1];
        dp[0] = 1;
        for (int i = 0; i < money.length; i++) {
            for (int j = money[i]; j < n + 1; j++){
                dp[j] += dp[j - money[i]];
            }
          }
        return dp[n];
    }
}

File: 프로그래머스/3/12907. 거스름돈/거스름돈.py
def solution(n, money):
    dp = [0] * (n + 1)
    dp[0] = 1
    
    for i in range(len(money)):
        for j in range(money[i], n + 1):
            dp[j] += dp[j - money[i]]
    
    answer = dp[n]
    return answer % 1000000007